components/layout/display_list_builder.rs

```diff
@@ -419,12 +419,35 @@ impl FragmentDisplayListBuilding for Fragment {
             // TODO: Check the bounds to see if a clip item is actually required.
             let clip = clip.clone().intersect_rect(&bounds);
 
+            // Background image should be positioned on the padding box basis.
+            let border = style.logical_border_width().to_physical(style.writing_mode);
+
+            // Use 'background-origin' to get the origin value.
+            let (mut origin_x, mut origin_y) = match background.background_origin {
+                background_origin::T::padding_box => {
+                    (Au(0), Au(0))
+                }
+                background_origin::T::border_box => {
+                    (-border.left, -border.top)
+                }
+                background_origin::T::content_box => {
+                    let padding = style.logical_padding().to_physical(style.writing_mode);
+                    let padding_left = model::specified(padding.left, bounds.size.width);
+                    let padding_top = model::specified(padding.top, bounds.size.height);
```

Comment:
Could you use `fragment.border_padding - border` instead of recomputing this from the style?